lib/shopify_app/session/in_memory_shop_session_store.rb

```diff
@@ -1,4 +1,7 @@
 module ShopifyApp
   class InMemoryShopSessionStore < InMemorySessionStore
+    def self.retrieve_by_shopify_domain(shopify_domain)
```

Comment:
This implementation doesn't seem right. The implementation in `InMemorySessionStore` stores by a random ID, but this expects to look up by `shopify_domain`. You may need to implement your own `self.store` here to do what you're expecting.